homeassistant/components/stream/__init__.py

```diff
@@ -572,6 +573,19 @@ def get_diagnostics(self) -> dict[str, Any]:
         """Return diagnostics information for the stream."""
         return self._diagnostics.as_dict()
 
+    @staticmethod
+    async def async_has_stream_auth_error(hass: HomeAssistant, source: str) -> bool:
+        """Return true if rtsp stream raises an HTTPUnauthorizedError error."""
+        import av  # pylint: disable=import-outside-toplevel
+
+        pyav_options = {"rtsp_flags": "prefer_tcp", "timeout": "5000000"}
+        try:
+            func = partial(av.open, source, options=pyav_options, timeout=5)
+            await hass.loop.run_in_executor(None, func)
+        except av.HTTPUnauthorizedError:
+            return True
+        return False
```

Comment:
I think it would be better to do the enum return of possible failures so its flexible enough to extend in the future without a breaking change since we expect other integrations to be able to call it.

Additionally, all API point should usually be a top level call it

`from homeassistant.components import stream`
`await stream.async_has_stream_auth_error(hass, source)`

Reply:
That's done